alibi_detect/od/knn.py

```diff
@@ -0,0 +1,48 @@
+from typing import Callable, Literal, Union, Optional
+import numpy as np
+
+from alibi_detect.od.base import OutlierDetector
+from alibi_detect.od.aggregation import BaseTransform
+
+from alibi_detect.od.backends import KNNTorch, KNNKeops
+from alibi_detect.utils.frameworks import BackendValidator
+
+backends = {
+    'pytorch': KNNTorch,
+    'keops': KNNKeops
+}
+
+
+class KNN(OutlierDetector):
+    def __init__(
+        self,
+        k: Union[int, np.ndarray],
+        kernel: Optional[Callable] = None,
+        aggregator: Union[BaseTransform, None] = None,
+        normaliser: Union[BaseTransform, None] = None,
+        backend: Literal['pytorch', 'keops'] = 'pytorch'
+    ) -> None:
+
+        backend = backend.lower()
+        BackendValidator(
+            backend_options={'pytorch': ['pytorch'],
+                             'keops': ['keops']},
+            construct_name=self.__class__.__name__
+        ).verify_backend(backend)
+
+        self.k = k
+        self.kernel = kernel
+        self.ensemble = isinstance(self.k, np.ndarray)
+        self.normaliser = normaliser
+        self.aggregator = aggregator
+        self.fitted = False
+        self.backend = backends[backend]
```

Comment:
It looks like `self.backend` here is the equivalent of `self._detector` for drift detectors. Any specific reason for the slightly different approach here?

Reply:
Essentially it is but `backend` here is just the score functionality and doesn't inherit from `BaseDetector`. I've added more notes here on [my thinking](https://www.notion.so/Hierarchy-055ebe62262c4ee78265bbae820addc7).